### PR TITLE
Fixed nativeq (Q) Switching Function Analytical Derivatives

### PR DESCRIPTION
**Pull request by nm3787**
##### Description

This fixes the derivative for the nativeq (Q) switching function in tools/SwitchingFunction.cpp. The derivative of the Q switching function should be multiplied by beta according to the chain rule and not divided by r0. This would affect the forces acting on CVs using this switching function. This modification was discussed with @hockyg.

##### Type of contribution

- [X] changes to code or doc authored by PLUMED developers, or additions of code in the core or within the default modules

##### Copyright

- [X] I agree to transfer the copyright of the code I have written to the PLUMED developers or to the author of the code I am modifying.

##### Tests

- [X] I added a new regtest or modified an existing regtest to validate my changes.
- [X] I verified that all regtests are passed successfully on [GitHub Actions](https://github.com/plumed/plumed2/actions).

### CODE REVIEW

File: src/tools/SwitchingFunction.cpp [hunk 1]
```diff
@@ -487,7 +487,7 @@ double SwitchingFunction::calculate(double distance,double&dfunc)const {
       dfunc=expression_deriv[t].evaluate();
     } else plumed_merror("Unknown switching function type");
 // this is for the chain rule:
-    dfunc*=invr0;
+    if(type!=nativeq) dfunc*=invr0;
```
Comment (carlocamilloni) [inline src/tools/SwitchingFunction.cpp:490]: Instead than this if I would divide for invr0 on line 454
if you can change this I will then merge it, thank!

**Comment by carlocamilloni** [comment]
@nm3787 thanks for reporting and fixing, if you can change the code as suggested I will then merge it

**Comment by hockyg** [comment]
> @nm3787 thanks for reporting and fixing, if you can change the code as suggested I will then merge it

Hi @carlocamilloni - I see why this makes sense from an efficiency point of view, but maybe you can add a code comment to warn people in case they add future switching functions?

**Comment by nm3787** [comment]
> @nm3787 thanks for reporting and fixing, if you can change the code as suggested I will then merge it

Thank you @carlocamilloni for the great suggestion. I have implemented it and added a comment as @hockyg suggested.

**Pull request merged**